Assets/Scripts/Interactables/Interactable.cs

```diff
@@ -93,12 +94,13 @@ public void FaceInteractable()
 
         // Animation of facing the interactable
         // Possible issue to prevent another coroutine being called if player is already rotating
-        StartCoroutine(RotateAnimation(player, rotation, player.GetComponent<PlayerController>().turnSpeed));
+        player.GetComponent<PlayerController>().CanMove = false;
+        StartCoroutine(RotateAnimation(player, rotation, player.GetComponent<PlayerController>().turnSpeed, () => UnfreezePlayer()));
 
     }
 
     // Coroutine that animates the rotation of the given object to the desiredRotation at a set turn speed
-    private IEnumerator RotateAnimation(GameObject obj, Quaternion desiredRotation, float turnSpeed)
+    protected IEnumerator RotateAnimation(GameObject obj, Quaternion desiredRotation, float turnSpeed, Action onFinishCallback)
```

Comment:
To make this function more general maybe we should see how to set a default null-ish value for `Action`, and just conditionally call it if its not null (so it's not necessary pass in a callback).

Reply:
Ok i'll make that a TODO for now if that's chill

Reply:
check discord!!!